### PR TITLE
chore(deployment): containerize `backend/` for production using Docker

### LINKED ISSUE #38: ⚙️ Setup: containerize `backend/` for production using `Docker`

### PR DESCRIPTION
**Pull request by jaysabe**
### Description

Created and added initial Dockerfile and .dockerignore for backend. 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)\
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes

### Issue ticket number and link

- Closes #38 

### Checklist

- [x] Self-reviewed my code prior to submission.
- [ ] Tests have been added or updated (if applicable).
- [ ] Documentation has been updated (if applicable).
- [ ] Screenshots added (if applicable).
- [x] This PR is ready for review.

### CODE REVIEW

**Comment by jaysabe** [comment]
looks like I still missed something in commitlint. Will follow up tomorrow on this.

**Comment by RickyC0626** [comment]
> looks like I still missed something in commitlint. Will follow up tomorrow on this.

Ah, looks like you can't have uppercase in commit due to convention.

File: backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# https://github.com/nodejs/docker-node?tab=readme-ov-file#image-variants
+FROM node:18.19.0-alpine
+
+# Set working directory in container
+WORKDIR /Project-Dahlia/backend
+
+# Copy dependencies lock to container
+COPY package.json package-lock.json ./
+
+# Set Yarn version
```
Comment (RickyC0626) [inline backend/Dockerfile]: **nitpick:** This is probably meant to be `npm`

File: backend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# syntax=docker/dockerfile:1
+
+# docker build -t backend .
+
+# Use the official Node runtime as parent image
+# https://github.com/nodejs/docker-node?tab=readme-ov-file#image-variants
+FROM node:18.19.0-alpine
```
Comment (RickyC0626) [inline backend/Dockerfile]: **suggestion:** To keep versioning consistent with CI, let's use `20.11.1-alpine`

**Review by RickyC0626 — CHANGES_REQUESTED** [review]

**Review by RickyC0626 — APPROVED** [review]

**Review by xamdoo — APPROVED** [review]

**Pull request merged**